ros_cross_compile/data_collector.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for time series data collection and writing said data to a file."""
+
+from contextlib import contextmanager
+from datetime import datetime
+from enum import Enum
+import json
+from pathlib import Path
+import time
+from typing import NamedTuple, Union
+
+from ros_cross_compile.sysroot_creator import INTERNALS_DIR
+
+
+Datum = NamedTuple('Datum', [('name', str),
+                             ('value', Union[int, float]),
+                             ('unit', str),
+                             ('timestamp', float),
+                             ('complete', bool)])
+
+
+class Units(Enum):
+    time = 'seconds'
```

Comment:
it's confusing to expose "measurables" rather than units. this should be seconds and bytes. python ints are essentially infinite size, so megabytes isn't necessary as you will not be limited to e.g. 4GiB